roundcube_structured_vacation_notice.php

```diff
@@ -148,37 +148,40 @@ public function on_message_objects_hook($args)
             false
         );
 
+       
+       
         /**
          * Only show structured email data if:
          * * our config flag allows it
          * * and the sender is a trusted one
          */
         if ($showStructuredEmailForTrustedSenders
+            && $contextType == "outOfOffice"
             && $rcmail->contact_exists($messageSender, $trustedSenderType)
-        ) {
-            // Set Dummy name & topic (no UI component yet)
-            $replacement_name = 'John Doe';
-            $replacement_topic = 'customer support';
-
-           // Decode jsonLd for our banner 
-           $decoded_jsonLd = json_decode($jsonLd);
-
-           $start = $decoded_jsonLd->start;
-           $end = $decoded_jsonLd->end;
-           $replacement_email = $decoded_jsonLd->replacement[0]->email;
-           // How we would access the name and topic if they were set (can't be set through ui currently)
-           //$replacement_name = $decoded_jsonLd->replacement[0]->name;           
-           //$replacement_topic = $decoded_jsonLd->replacement[0]->topic;
+            ) 
+        {
+            
+            // Decode jsonLd for our banner 
+            $decoded_jsonLd = json_decode($jsonLd);
+            
+        
+            $replacement_name = $decoded_jsonLd->replacement[0]->name;
+            $replacement_topic = $decoded_jsonLd->replacement[0]->topic;
+
+            $start = $decoded_jsonLd->start->date;
+            $end = $decoded_jsonLd->end->date;
+            $replacement_email = $decoded_jsonLd->replacement[0]->email;
             
             // OUTPUT Banner similar to the "remote-objects-message"
-            // TODO only show the parameters which are set
+
```

Comment:
It will render only the parameters which are not null at the string templating below